user_tools/tests/spark_rapids_tools_ut/test_tool_argprocessor.py

```diff
@@ -74,112 +74,282 @@ def validate_args_w_savings_disabled(tool_name: str, t_args: dict):
+    def test_with_cluster_name_with_eventlogs(self, get_ut_data_dir, tool_name):
+        # should fail: defaults to onprem, cannot run when the cluster is by name
+        self.create_tool_args_should_fail(tool_name,
+                                          cluster='my_cluster',
+                                          eventlogs=f'{get_ut_data_dir}/eventlogs')
 
     @pytest.mark.parametrize('tool_name', ['qualification', 'profiling'])
-    @pytest.mark.parametrize('csp', csps)
-    @register_triplet_test([ArgValueCase.VALUE_A, ArgValueCase.VALUE_A, ArgValueCase.UNDEFINED])
-    def test_cluster_name_no_eventlogs(self, tool_name, csp):
-        # Missing eventlogs should be accepted for all CSPs (except onPrem)
-        # because the eventlogs can be retrieved from the cluster
-        tool_args = AbsToolUserArgModel.create_tool_args(tool_name,
-                                                         platform=csp,
-                                                         cluster='my_cluster')
-        assert tool_args['runtimePlatform'] == CspEnv(csp)
-        self.validate_args_w_savings_enabled(tool_name, tool_args)
+    @pytest.mark.parametrize('csp', all_csps)
+    @register_triplet_test([ArgValueCase.IGNORE, ArgValueCase.IGNORE, ArgValueCase.IGNORE])
+    @register_triplet_test([ArgValueCase.IGNORE, ArgValueCase.IGNORE, ArgValueCase.VALUE_A])
+    @register_triplet_test([ArgValueCase.IGNORE, ArgValueCase.VALUE_A, ArgValueCase.IGNORE])
+    @register_triplet_test([ArgValueCase.IGNORE, ArgValueCase.VALUE_A, ArgValueCase.VALUE_A])
+    @register_triplet_test([ArgValueCase.VALUE_A, ArgValueCase.IGNORE, ArgValueCase.IGNORE])
+    @register_triplet_test([ArgValueCase.VALUE_A, ArgValueCase.IGNORE, ArgValueCase.VALUE_A])
+    @register_triplet_test([ArgValueCase.VALUE_A, ArgValueCase.VALUE_A, ArgValueCase.IGNORE])
+    @register_triplet_test([ArgValueCase.VALUE_A, ArgValueCase.VALUE_A, ArgValueCase.VALUE_A])
+    def test_with_platform_with_cluster_name_with_eventlogs(self, get_ut_data_dir, tool_name, csp):
+        if CspEnv(csp) != CspEnv.ONPREM:
+            # should pass: platform, cluster name and eventlogs are provided
+            tool_args = self.create_tool_args_should_pass(tool_name,
+                                                          platform=csp,
+                                                          cluster='my_cluster',
+                                                          eventlogs=f'{get_ut_data_dir}/eventlogs')
+            self.validate_tool_args(tool_name=tool_name, tool_args=tool_args,
+                                    cost_savings_enabled=True,
+                                    expected_platform=csp)
+        else:
+            # should fail: onprem platform cannot run when the cluster is by name
+            self.create_tool_args_should_fail(tool_name,
+                                              platform=csp,
+                                              cluster='my_cluster',
```

Comment:
In register_triplet_test, for the case of cluster =IGNORE, if it's not defined(UNDEFINED) then the test should pass for on prem cluster, right? i.e based on the values of platform and eventlogs.

Reply:
Thanks @nartal1.  I have refactored the code to remove all the IGNORE cases. As we had discussed, there are only 16 cases needed. I have updated the description with this.